eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Imp/UtxoSpec.hs

```diff
@@ -63,6 +78,34 @@ spec =
         [fromNativeScript spendingScript, fromNativeScript spendingScript]
           ++ extraScripts
           ++ extraScripts
+    describe "disjoint inputs and reference inputs" $ do
+      let
+        scriptHash lang = hashPlutusScript $ inputsOverlapsWithRefInputs lang
+        mkTestTx :: TxIn -> Tx era
+        mkTestTx txIn =
+          mkBasicTx mkBasicTxBody
+            & bodyTxL . inputsTxBodyL .~ Set.singleton txIn
+            & bodyTxL . referenceInputsTxBodyL .~ Set.singleton txIn
+
+      it "Cannot run scripts that expect inputs and refInputs to overlap (PV 9)" $ whenMajorVersion @9 $ do
+        txIn <- produceScript $ scriptHash SPlutusV3
+        submitFailingTx @era
+          (mkTestTx txIn)
+          [ injectFailure $ BabbageNonDisjointRefInputs [txIn]
+          ]
+      it "Same script cannot appear in regular and reference inputs in PlutusV3 (PV 10)" $ whenMajorVersion @10 $ do
+        txIn <- produceScript $ scriptHash SPlutusV3
+        submitFailingTx @era
+          (mkTestTx txIn)
+          [ injectFailure $ BabbageNonDisjointRefInputs [txIn]
+          ]
```

Comment:
It is cleaner to combine two identical tests into one
```suggestion
      it "Cannot run scripts that expect inputs and refInputs to overlap (PV 9/10)" $ do 
        whenMajorVersionAtMost @10 $ do
          txIn <- produceScript $ scriptHash SPlutusV3
          submitFailingTx @era
            (mkTestTx txIn)
            [ injectFailure $ BabbageNonDisjointRefInputs [txIn]
            ]
```

FYI. We could add `whenMajorVersionAtLeast @9`, but that would be redundant, since Conway starts with `9` and this test is only valid starting with Conway